R/QFeatures-constructors.R

```diff
@@ -1,85 +1,233 @@
+##'     acquisition batch (given by `batchCol`) and the acquisition
+##'     channel within the batch (e.g. TMT channel, given by
+##'     `channelCol`). Additional fields (e.g. sample type,
+##'     acquisition date,...) are allowed and will be stored as sample
+##'     meta data.
 ##'
-##' @author Laurent Gatto
+##' @param fnames For the single- and multi-set cases, an optional
+##'     `character(1)` or `numeric(1)` indicating the column to be
+##'     used as feature names. If missing, in the latter case, the
+##'     rows are named PSM1, PSM2, PSM3, ... Note that rownames must
+##'     be unique in `QFeatures` sets.
 ##'
-##' @describeIn readQFeatures See description.
+##' @param name For the single-set case, an optional `character(1)` to
+##'     name the set in the `QFeatures` object. If not set, `psms`
+##'     is used.
+##'
+##' @param batchCol For the multi-set case, a `numeric(1)` or
+##'     `character(1)` pointing to the column of `assayData` and
+##'     `colAnnotation` that contain the batch names. Make sure that
+##'     the column name in both table are either identical and
+##'     syntactically valid (if you supply a `character`) or have the
+##'     same index (if you supply a `numeric`). Note that characters
+##'     can be converted to syntactically valid names using
+##'     `make.names`
+##'
+##' @param channelCol For the multi-set case, a `numeric(1)` or
+##'     `character(1)` pointing to the column of `colData` that
+##'     contains the column names of the quantitative data in
+##'     `assayData` (see Example).
+##'
+##' @param suffix For the multi-set case, a `character()` giving the
+##'     suffix of the column names in each set. Sample/single-cell
+##'     (column) names are automatically generated using: batch name +
+##'     sep + suffix. Make sure suffix contains unique character
+##'     elements. The length of the vector should equal the number of
+##'     quantification channels.  If `NULL` (default), the suffix is
+##'     derived from the the names of the quantification columns in
+##'     `assayData`.
+##'
+##' @param sep A `character(1)` that is inserted between the set
+##'     name and the `suffix` (see `suffix` argument for more
+##'     details).
+##'
+##' @param removeEmptyCols A `logical(1)`. If true, the function will
+##'     remove in each batch the columns that contain only missing
+##'     values.
+##'
+##' @param ecol Same as `colAnnotation` for the single-set
+##'     case. Available for backwards compatibility. Default is
+##'     `NULL`. If both `ecol` and `colAnnotation` are set, an error
+##'     is thrown.
+##'
+##' @param verbose A `logical(1)` indicating whether the progress of
+##'     the data reading and formatting should be printed to the
+##'     console. Default is `TRUE`.
+##'
+##' @return An instance of class `QFeatures` or
+##'     [SummarizedExperiment::SummarizedExperiment()]. For the
+##'     former, the expression data of each batch is stored in a
```

Comment:
"expression data", I think this is still an artefact from `MSnbase` and the `ExpressionSet` class :grin:. For consistency, maybe use "assay data"  or "feature data". Also maybe use "each run" instrad of "each batch" (cf comment above).